src/LanguageServer/Microsoft.CodeAnalysis.LanguageServer/BrokeredServices/ServiceBrokerFactory.cs

```diff
@@ -55,6 +56,13 @@ public ServiceBrokerFactory([ImportMany] IEnumerable<IOnServiceBrokerInitialized
     /// </summary>
     public IServiceBroker? TryGetFullAccessServiceBroker() => _container?.GetFullAccessServiceBroker();
 
+    /// <summary>
+    /// Returns a task for full-access service broker. Task might never complete, e.g. if Dev Kit is not available.
```

Comment:
Since we should know at start whether we're ever going to get a service broker or not, can we just make this nullable Task<> so it's easier to deal with that case? Or does this actually make it easier to deal with?